api/types.proto

```diff
@@ -86,6 +86,18 @@ message NodeDescription {
 
 	// Information about the Docker Engine on the node.
 	EngineDescription engine = 4;
+
+	// Information on the node's TLS setup
+	NodeTLSInfo tls_info = 5 [(gogoproto.customname) = "TLSInfo"];
+}
+
+message NodeTLSInfo {
+	// Information about which root certs the node trusts
+	string trust_root_hash = 1;
```

Comment:
Do you mean bytes on the `SessionMessage` vs the hash here?  No particular reason to use the hash instead - I can just send the bytes.  But on the `SessionMessage` it has to be bytes because that's how the node gets the new root certificate to trust.